lua/plenary/window/border.lua

```diff
@@ -189,7 +181,29 @@ function Border:new(content_bufnr, content_win_id, content_win_options, border_w
     width = content_win_options.width + thickness.left + thickness.right,
     height = content_win_options.height + thickness.top + thickness.bot,
     noautocmd = content_win_options.noautocmd,
-  })
+  }
+  if create_window then
```

Comment:
I'd factor out a `_get_config` helper (resolving user args, defaults, and config of existing window) for `new` and `set_size` as opposed to creating the window from `set_size` which appears a bit confusing. Thoughts?

Reply:
I agree that the `create_window` stuff was a bit weird, so I have factored out the overlapping parts for creating/resizing borders and just kept the creating/resizing of border window in the relevant function.
I think that checking for existence of `win_id` would probably be confusing behaviour for some users, so completely separate functions made more sense to me.

Let me know if you are happy with the new setup 🙂

Reply:
Looks great! `__update_lines_get_win_config` naming itches me a bit though. You don't really get a config (even more so by having to pass configs) but rather conform content and border window configs. I don't know, I like the simple `__align(content_win_options, border_win_options)` better, with comments that lines and window configs are "aligned".

This is very nitty though. If you are rolling your eyes too much at this just keep it :sweat_smile:

Reply:
Yeh, I agree that the current name is a bit too long.
However, just naming it `__align` doesn't suggest that it returns a conformed config table.
Maybe something like `__align_calc_config` or `__align_conform_config`?

Reply:
Up to you :) I think you can ping Conni afterwards and get this merged :rocket: